akka-remote/src/main/scala/akka/remote/artery/Association.scala

```diff
@@ -1188,7 +1195,8 @@ private[remote] class AssociationRegistry(createAssociation: Address => Associat
     currentMap.keysIterator.foreach { uid =>
       val association = currentMap.get(uid).get
       val state = association.associationState
-      if (state.isQuarantined() && ((now - state.lastUsedTimestamp.get) >= afterNanos))
+      if ((state.isQuarantined() || !state
+            .isHandshakeCompleted()) && ((now - state.lastUsedTimestamp.get) >= afterNanos))
```

Comment:
same as previous:

```scala
if (state.addressState != AddressWithMissingUID 
    && ((now - state.lastUsedTimestamp.get) >= afterNanos))
```

Maybe we don't even need a three state. All that we are trying to detected is if we are in the case of having an address but missing its UID. Maybe once single method with a clear name will do the trick.